anvil/src/udev.rs

```diff
@@ -133,20 +151,68 @@ pub fn run_udev(log: Logger) {
     /*
      * Initialize the compositor
      */
+    let primary_gpu = primary_gpu(&session.seat())
+        .unwrap()
+        .and_then(|x| DrmNode::from_path(x).ok()?.node_with_type(NodeType::Render)?.ok())
+        .unwrap_or_else(|| {
+            all_gpus(&session.seat())
+                .unwrap()
+                .into_iter()
+                .find_map(|x| DrmNode::from_path(x).ok())
+                .expect("No GPU!")
+        });
+    info!(log, "Using {} as primary gpu.", primary_gpu);
+
+    let mut gpus = GpuManager::new(EglGlesBackend, log.clone()).unwrap();
+    #[cfg_attr(not(feature = "egl"), allow(unused_mut))]
+    let mut renderer = gpus
+        .renderer::<Gles2Renderbuffer>(&primary_gpu, &primary_gpu)
+        .unwrap();
+
+    #[cfg(feature = "egl")]
+    {
+        info!(
+            log,
+            "Trying to initialize EGL Hardware Acceleration via {:?}", primary_gpu
+        );
+        if renderer.bind_wl_display(&*display.borrow()).is_ok() {
+            info!(log, "EGL hardware-acceleration enabled");
+        }
+    }
+
+    #[cfg(feature = "debug")]
+    let fps_image =
+        image::io::Reader::with_format(std::io::Cursor::new(FPS_NUMBERS_PNG), image::ImageFormat::Png)
+            .decode()
+            .unwrap();
+    #[cfg(feature = "debug")]
+    let fps_texture = renderer
+        .import_memory(
+            &fps_image.to_rgba8(),
+            (fps_image.width() as i32, fps_image.height() as i32).into(),
+            false,
+        )
+        .expect("Unable to upload FPS texture");
+
     #[cfg(feature = "egl")]
-    let primary_gpu = primary_gpu(&session.seat()).unwrap_or_default();
+    let dmabuf_formats = renderer.dmabuf_formats().cloned().collect::<Vec<_>>();
 
     // setup the timer
     let timer = Timer::new().unwrap();
 
+    std::mem::drop(renderer);
```

Comment:
Yes, because the renderer borrows `gpus` mutably, so without the drop rustc complains, that it cannot move `gpus` into the `UdevData` in the next line.

Reply:
Added a comment to explain this.

Reply:
Hmm, maybe some recent changes made this obsolete? I will remove it and see what CI think about it.